### ISSUE TITLE
Parse errors when conditional directives are combined with raw string literals

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#69776]
```csharp
#if false

var x = """
    # Hello raw string literals
    """;
    
#endif
```

I can see why that fails, but I'm wondering if this can be improved. I think that when the compiler encounters a conditional directive that evaluates to false, it should try to keep going until it finds a valid directive trivia.

So instead of parsing "# Hello raw string literals" with as "BadDirectiveTrivia", it should be parsed as DisabledTextTrivia and continue searching for a valid directive trivia.

cc @CyrusNajmabadi

**Comment by Youssef1313** [comment]
This also happens with the verbatim strings.

The ambiguity will still be there anyway if the verbatim/raw string contains syntactically valid directives though :/

**Comment by CyrusNajmabadi** [comment]
This is a request for a language change.  The compiler is working as per the language rules here.  I'm wary about any changes here because of hte potential for regressions.

**Comment by Youssef1313** [comment]
Is the current behavior well-spec'ed? If so, can you move this as a discussion to csharplang?

**Comment by CyrusNajmabadi** [comment]
> Is the current behavior well-spec'ed? If so, can you move this as a discussion to csharplang?

Yes, it's spec'ed here: https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/lexical-structure.md#65-pre-processing-directives
